logic.js

```diff
@@ -0,0 +1,93 @@
+const Tweeter = () => {
+    const _posts = [{
+            text: "First post!",
+            id: "p1",
+            comments: [
+                { id: "c1", text: "First comment on first post!" },
+                { id: "c2", text: "Second comment on first post!!" },
+                { id: "c3", text: "Third comment on first post!!!" }
+            ]
+        },
+        {
+            text: "Aw man, I wanted to be first",
+            id: "p2",
+            comments: [
+                { id: "c4", text: "Don't wory second poster, you'll be first one day." },
+                { id: "c5", text: "Yeah, believe in yourself!" },
+                { id: "c6", text: "Haha second place what a joke." }
+            ]
+        }
+    ];
+    let postIdCounter = () => _posts.length;
+    let commentIdCounter = _posts.map(post => post.comments.length).reduce((a, b) => a + b);
+
+    const getPosts = () => _posts;
+
+    const addPost = text => _posts.push({ text: text, id: "p" + parseInt(postIdCounter() + 1), comments: [] });
+
+    const removePost = postId => {
+        for (let post in _posts) {
```

Comment:
for the for-in loop, you should call the variable "i" or "index", and not "post" -> because it indicated the index and not the post itself.